cpp/src/groupby/hash/create_output.cu

```diff
@@ -0,0 +1,232 @@
+/*
+ * Copyright (c) 2024-2025, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "create_output.hpp"
+#include "helpers.cuh"
+
+#include <cudf/aggregation.hpp>
+#include <cudf/column/column_factories.hpp>
+#include <cudf/detail/aggregation/aggregation.hpp>
+#include <cudf/dictionary/dictionary_column_view.hpp>
+#include <cudf/null_mask.hpp>
+#include <cudf/table/table_view.hpp>
+#include <cudf/types.hpp>
+#include <cudf/utilities/span.hpp>
+
+#include <rmm/cuda_stream_view.hpp>
+#include <rmm/device_uvector.hpp>
+
+#include <cuco/static_set.cuh>
+#include <thrust/scatter.h>
+#include <thrust/tabulate.h>
+#include <thrust/transform.h>
+
+namespace cudf::groupby::detail::hash {
+namespace {
+
+/**
+ * @brief Functor to create sparse result columns for hash-based groupby aggregations
+ *
+ * This functor handles the creation of appropriately typed and sized columns for each
+ * aggregation, including special handling for SUM_WITH_OVERFLOW which requires a struct column.
+ */
+struct sparse_column_creator {
```

Comment:
is the `sparse_column_creator` name still approriate? I thought we are creating dense output here